patroni/postgresql.py

```diff
@@ -1367,16 +1369,16 @@ def save_configuration_files(self):
         try:
             for f in self._configuration_to_save:
                 if os.path.isfile(f):
-                    shutil.copy(f, f + '.backup')
+                    shutil.copy(f, f.replace(self._config_dir, self._data_dir) + '.backup')
```

Comment:
This `replace` scares me.
I think it's better to be explicit, i.e. make `_configuration_to_save` return non-absolute file names ('postgresql.conf', 'pg_hba.conf', etc...) and copy them between `config_dir` and `data_dir`.

Everything else looks good to me.